vars/neoDeploy.groovy

```diff
@@ -216,11 +231,80 @@ void call(parameters = [:]) {
                 lock("$STEP_NAME:${neoCommandHelper.resourceLock()}") {
                     deploy(script, configuration, neoCommandHelper, configuration.dockerImage, deployMode)
                 }
+                if(configuration.neo.invalidateCache == true) {
+                    invalidateCache(configuration)
+                }
             }
         }
     }
 }
 
+private invalidateCache(configuration){
+    def account = configuration.neo.account
+    def host = configuration.neo.host
+
+    withCredentials([usernamePassword(
+        credentialsId: configuration.neo.oauthCredentialId,
+        passwordVariable: 'OAUTH_NEO_CLIENT_SECRET',
+        usernameVariable: 'OAUTH_NEO_CLIENT_ID')]) {
+        def bearerTokenResponse = sh(
+            script: """#!/bin/bash
+                    curl -X POST -u "${OAUTH_NEO_CLIENT_ID}:${OAUTH_NEO_CLIENT_SECRET}" \
+                    --fail \
+                    "https://oauthasservices-${account}.${host}/oauth2/api/v1/token?grant_type=client_credentials&scope=write,read"
+                    """, returnStdout: true)
+        def fetchBearerTokenJson = readJSON text: bearerTokenResponse
+
+        def bearerToken = fetchBearerTokenJson.access_token
```

Comment:
as we are only interested in the token:
```suggestion
        def bearerToken = (readJSON text: bearerTokenResponse).access_token
```

Reply:
This syntax does not work.

Reply:
sorry about that! I updated the suggestion.